### DISCUSSION TITLE
"Bool value of tensor with more than one value is ambiguous" error when using "dp" but not "ddp"?

### DISCUSSION THREAD

**Discussion by EvanZ** [Lightning-AI/torchmetrics#878] [General]
So I'm getting the following error which is not easy to track down:

`"Bool value of tensor with more than one value is ambiguous"`

It is throwing for the line of code that is doing the fitting:

`trainer.fit(model, datamodule=dm)`

This code has worked before on a cpu, but this is the first time I'm testing it on a GPU instance (a single 4-GPU instance running on EC2). It seems that this error occurs when I use 'dp' but runs without error when I switch to 'ddp'. Is this expected behavior?

**Comment by SkafteNicki** [comment]
It does not sound like an expected error.
If you could provide the full traceback we can better help you debug
